pallets/idn-consumer/src/traits.rs

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2025 by Ideal Labs, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Pallet IDN Consumer traits
+
+use sp_idn_traits::pulse::Pulse;
+
+/// A trait for describing a pulse consumtion behavior.
+pub trait PulseConsumer<P: Pulse, I, O, E> {
+	fn consume_pulse(pulse: P, sub_id: I) -> Result<O, E>;
+}
+
+/// A trait for describing a quote consumtion behavior.
+pub trait QuoteConsumer<Q, E, O> {
+	fn consume_quote(quote: Q) -> Result<O, E>;
+}
+
+/// A trait for describing a subscription info consumtion behavior.
+pub trait SubInfoConsumer<S, E, O> {
```

Comment:
For consistency, it would be good to have the `E` and `O` types in the same order in both QuoteConsumer and PulseConsumer.